docs/references/function_calling.md

```diff
@@ -0,0 +1,61 @@
+# Tools Calling
```

Comment:
Merge `docs/references/function_calling.md` and `docs/backend/function_tooling.ipynb`.
I think we only need the content of this file but in the format of an interactive notebook.

The outline of this doc is very good!